src/Data/Hermes.hs

```diff
@@ -544,6 +546,15 @@ getDouble valPtr = withRunInIO $ \run ->
     liftIO $ peek ptr
 {-# INLINE getDouble #-}
 
+getDoubleNonFinite :: Value -> Decoder Double
+getDoubleNonFinite = withRawByteString $ \bs ->
+  case BSC.strip bs of
+    "\"+inf\"" -> pure $ 1/0
+    "\"-inf\"" -> pure $ (-1)/0
+    "null"     -> pure $ 0/0
+    _          -> Sci.toRealFloat <$> parseScientific bs
```

Comment:
Thanks for quickly developing this PR.

I noticed that it matches on the actual bytestring inside the JSON AST. This means that if the string is encoded e.g. using escape codes, it wouldn't parse. Granted, this is a very academic issue, but I thought it was worth pointing out.

For example, using Aeson:

```
ghci> decode @Double "\"\\u002dinf\""
Just (-Infinity)
```

But using this PR:

```
ghci> decodeEither personDecoder "{\"lol\": \"\\u002dinf\"}"
Left (InternalException (HError {path = "/lol", errorMsg = "Failed to parse Scientific: Failed reading: takeWhile1", docLocation = "\"\\u002dinf\"}", docDebug = "json_iterator [ depth : 2, structural : '\"', offset : 8', error : No error ]"}))
ghci> decodeEither personDecoder "{\"lol\": \"-inf\"}"
Right (Person (-Infinity))
```

I don't really mind too much, I think the PR is fine as is. I don't think it is likely that people will re-encode their Aeson JSON such that it uses escape codes for the non-finite IEEE 754 values.

Reply:
Yeah that's the trick with `get_raw_json_token`, whereas `get_string` escapes the string but will generate an exception if the value isn't actually a string. I will see if I can address this without losing performance. Although I do agree not handling escaped strings for this particular decoder should not be a major problem.